get-extracts/get-extracts.py

```diff
@@ -1,146 +1,81 @@
-#!/usr/bin/env python
 
-# Created by Brian Turley
-# Solutions Engineer
 # Ookla
-# Updated 1/30/17
+# Updated 5/30/18
 
 # This Python script queries a list of available data extract files from Speedtest Intelligence,
 # determines what data sets are available, and then downloads the most recent version of each.
 # By default the files are stored in the directory where the script is running, but modifying
 # the storageDir variable will allow you to specify a directory.
 
-import urllib2
+try: # Python3
+    import urllib.request as compatible_urllib
+except ImportError: # Python 2
+    import urllib2 as compatible_urllib
+
 import json
 import os
 import base64
 import sys
 
+extracts_url = 'https://intelligence.speedtest.net/extracts'
 
-theurl = 'https://intelligence.speedtest.net/extracts'
-
-#Please replace MyApiKey and MyApiSecret below with your
-#organization's API key.
-
-username = 'MyApiKey'
-password = 'MyApiSecret'
-
-
-
-#By default, the script stores the extract files in the directory where the script is running
-#To specify a storage directory, change this value to a string represting the directory where
-#the files should be stored.
-#Example: storageDir = '/data/ookla/extracts'
+# Please replace MyApiKey and MyApiSecret below with your organization's API key.
+username = 'my_api_key'
+password = 'my_api_secret'
 
+# By default, the script stores the extract files in the directory where the script is running
+# To specify a storage directory, change this value to a string represting the directory where
+# the files should be stored.
+# Example: storageDir = '/data/ookla/extracts'
 storageDir = os.getcwd()
 
-opener = urllib2.build_opener()
-urllib2.install_opener(opener)
-
-#request json list of files
+opener = compatible_urllib.build_opener()
+compatible_urllib.install_opener(opener)
 opener.addheaders = [('Accept', 'application/json')]
 
-#setup authentication
-base64string = base64.b64encode('%s:%s' % (username, password))
+# setup authentication
+login_credentials = '%s:%s' % (username, password)
+base64string = base64.b64encode(login_credentials.encode('utf-8')).decode('ascii')
 opener.addheaders = [('Authorization', 'Basic %s' % base64string)]
 
-#If login page is returned, raise error message.
-result = urllib2.urlopen(theurl)
-if result.info().type == "text/html":
-    print "Authentication error.\nPlease verify that the API key is correct."
-    sys.exit()
-
+# makes request for files
+response = compatible_urllib.urlopen(extracts_url).read()
 try:
-    response = result.read()
-except urllib2.HTTPError, err:
-   if err.code == 500:
-       print "Error: The account associated with this API key does not have access to data extracts.\nPlease contact your technical account manager to enable data extracts for this account."
```

Comment:
Did we lose these error cases? It seems like we don't print friendly messages for `HTTP 500` or `Content-Type: text/html` like we did before.
Admittedly, assuming that all `text/html` responses are authentication errors seems a little odd, but we should still be able to provide the user a nice error message if they don't have proper API Keys, ya?

Reply:
The urlib 2 or urlib automatically prints errors. But any 500 error is not a result of an API key issue (that will cause a 401) but some bug with ookla-netmetrics or the STI-server. I'll look into this more and try to come up with helpful error messages.